ui/backend/client_ui.py

```diff
@@ -131,19 +131,23 @@ def update_similar_snippets(self,
             The number of similar snippets found and updated
         """
 
-        discoveries = self.get_discoveries(repo_url,
-                                           file_name,
-                                           state_filter='new')[1]
-        model = build_embedding_model()
-        target_snippet_embedding = compute_snippet_embedding(target_snippet,
-                                                             model)
+        discoveries = self.get_discoveries(repo_url, file_name)[1]
```

Comment:
this method returns a list of dictionaries. Why are you taking [1]? if you don't have discoveries (or if you have only 1) this should raise a IndexError

Reply:
No, in the UI classes, this method returns the number of discoveries as 1st element and the actual discoveries as 2nd element